ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java

```diff
@@ -104,9 +105,12 @@ private static PreContingencyResult buildPreContingencyResult(Leaf rootLeaf, Lea
         });
 
         // preventive Network Actions
-        optimalLeaf.getNetworkActions().forEach(na -> {
-            remedialActionResultList.add(new RemedialActionResult(na.getId(), na.getName(), true, buildRemedialActionElementResult(na)));
-        });
+        optimalLeaf.getNetworkActions().forEach(na -> remedialActionResultList.add(
+                new RemedialActionResult(na.getId(), na.getName(), true, buildRemedialActionElementResult(na))));
+
+        // preventive Range Actions
+        crac.getRangeActions().forEach(rangeAction -> remedialActionResultList.add(
+                new RemedialActionResult(rangeAction.getId(), rangeAction.getName(), true, getRemedialActionElementResult(rangeAction, optimalLeaf))));
```

Comment:
You cannot put all the range actions from the Crac in the results.
Only the one selected by the optimalLeaf

something like optimalLeaf.getRaoComputationResults().getPreContingency().getRemedialActionResults()

Reply:
indeed!